esp8266/modesp.c

```diff
@@ -35,6 +35,492 @@
 #include MICROPY_HAL_H
 #include "queue.h"
 #include "user_interface.h"
+#include "espconn.h"
+
+#include "utils.h"
+
+#define IPV4ADDR_BUFSIZE    4
+
+// Helpers from cc3200/mods/modnetwork.c
+STATIC void esp_espconn_parse_ipv4_addr(mp_obj_t addr_in, uint8_t *out_ip) {
```

Comment:
+1 I understand such refactoring is a bit "hard" to do for contributors (vs maintainers), but duplicating code is the worst choice. If not complete factoring out those funcs from cc3200 code, then at least linking to them (they may be part of bigger module, so that may be hard). @atalax , so let us know if you feel like giving that a try or if we should do it first. If you try, the good location for those functions is lib/ .